src/Sarif/Writers/CachingLogger.cs

```diff
@@ -48,6 +48,11 @@ public void Log(ReportingDescriptor rule, Result result)
                 throw new ArgumentNullException(nameof(result));
             }
 
+            if (!ShouldLog(result))
```

Comment:
>ShouldLog [](start = 17, length = 9)

Ouch! Fantastic find, well done! This small change will make an enormous difference.